gamemodes/terrortown/entities/entities/ttt_basegrenade_proj.lua

```diff
@@ -56,28 +56,31 @@ end
 ---
 -- @ignore
 function ENT:Think()
-    local etime = self:GetExplodeTime() or 0
-    if etime ~= 0 and etime < CurTime() then
+    local timeExplosion = self:GetExplodeTime() or 0
+
+    if timeExplosion ~= 0 and timeExplosion < CurTime() then
+        local thrower = self:GetThrower()
+
         -- if thrower disconnects before grenade explodes, just don't explode
-        if SERVER and (not IsValid(self:GetThrower())) then
+        if SERVER and (not IsValid(thrower)) then
             self:Remove()
-            etime = 0
+
             return
         end
 
         -- find the ground if it's near and pass it to the explosion
-        local spos = self:GetPos()
+        local pos = self:GetPos()
         local tr = util.TraceLine({
-            start = spos,
-            endpos = spos + Vector(0, 0, -32),
-            mask = MASK_SHOT_HULL,
-            filter = self.thrower,
+            start = pos,
+            endpos = pos + Vector(0, 0, -32),
+            mask = MASK_SOLID_BRUSHONLY,
```

Comment:
This removes quite a lot from the mask (which is probably a good thing)

afaik this is essentially only CONTENTS_SOLID and CONTENTS_MOVEABLE [1] which certainly seems fine, but we should extensively test this

[1] https://wiki.facepunch.com/gmod/Enums/CONTENTS

Reply:
I agree with you here. However at least all players and the entity itself should be excluded here. That would be the alternative. Maybe we should test this on more maps